src/gui/torrentcontentmodel.cpp

```diff
@@ -551,20 +564,47 @@ void TorrentContentModel::setupModelData(const BitTorrent::AbstractFileStorage &
     emit layoutChanged();
 }
 
-void TorrentContentModel::selectAll()
+void TorrentContentModel::notifyModelUpdate(const QModelIndex &index)
```

Comment:
Its purpose is purely specific for those cases when it is necessary to notify about all affected item priorities changes when the priority of some item is changed. I would use a more highly specialized name than "model update". Maybe `notifyPriorityChanged()`? Or `notifyCheckStateChanged()`?

Reply:
This function currently notify the whole model. A finer notification should be possible but I haven't thought thoroughly about it and I think it belongs to another PR.

Reply:
>it is also utilized by other column updates

What other column (except related to download priority) update needs to affect rows other than directly changed one?

Reply:
>What other column (except related to download priority) update needs to affect rows other than directly changed one?

You can see in code that some other functions should utilize this new implementation.
I plan to keep this function as is for now and in a later PR, it would be possible to add a range parameter to this function that allow specifying columns. The function would still need a general name in the end.
The reason I don't want to do it now is because it will be more involved with legacy code which might become a deep rabbit hole, After all this PR is intended to solve an issue and the follow-up PR can be an optimization/generalization.

Reply:
> You can see in code that some other functions should utilize this new implementation.

Only ones that need to propagate changes in some row to another rows, aren't they?

Reply:
> So tree traversal is required in any case of notifying/updating.

Really? This is quite expected when the priority of some file/folder is changed, so it should be extended to parent and child files/folders. But I wouldn't expect any tree traversal in case when  user change name of some file.
Unfortunately, I have now remembered about ugly implementation of "content model", so it is most likely not organized currently as it should ideally be.

Reply:
>But I wouldn't expect any tree traversal in case when user change name of some file.

This is already handled properly...

Reply:
> > But I wouldn't expect any tree traversal in case when user change name of some file.
> 
> Yes, and this is already handled properly.

How it is handled? I suppose by not using `notifyModelUpdate()` in similar cases, yes?

Reply:
>I suppose by not using notifyModelUpdate() in similar cases, yes?

Yes, by emitting `dataChanged()` directly.

Reply:
So `notifyModelUpdate()` isn't still intended for general purpose but only for the cases where changing of data in some cell (identified by index) should be recursively propagated to parent/child rows. But nothing in its signature indicates this specificity, which gives future contributors a reason to use it for incorrect purposes (i.e. in cases where such recursive notifications are not required).
So why not name it more specifically? Since it emits `dataChanged()` signal recursively it can be named `notifyDataChangedRecursively()`. Then it will be self-descriptive, showing its relationship with both `dataChanged()` signal and its recursive specifics.

Reply:
>Since it emits dataChanged() signal recursively it can be named notifyDataChangedRecursively(). Then it will be self-descriptive, showing its relationship with both dataChanged() signal and its recursive specifics.

I don't think 'recursive' can be linked to traversing up and down of a tree. The word means something else for me.

Reply:
> I don't think 'recursive' can be linked to traversing up and down of a tree. The word means something else for me.

I agree, this is not an accurate use of this term. Recursion is just one way to traverse the tree. I've just used it by analogy.
But your claims only affect the proposed solution. The problem remains in place. 

Besides, you can just merge this PR, leaving this problem to the successors.

Reply:
Some random ideas, what about `notifyCellFamiliyTreeUpdate()`, `notifyRelatedRowUpdate()` or `notifyPartialTreeUpdate()`?

>Besides, you can just merge this PR, leaving this problem to the successors.

OK, just here for discussion because I have a follow-up PR.

Reply:
`notifyPartialTreeUpdated()` looks good enough (or maybe even better
`notifySubtreeUpdated()`).